src/java.base/share/classes/jdk/internal/vm/VMSupport.java

```diff
@@ -51,32 +56,56 @@ public static synchronized Properties getAgentProperties() {
     private static native Properties initAgentProperties(Properties props);
 
     /**
-     * Write the given properties list to a byte array and return it. Properties with
-     * a key or value that is not a String is filtered out. The stream written to the byte
-     * array is ISO 8859-1 encoded.
+     * Writes the given properties list to a byte array and return it. The stream written
+     * to the byte array is ISO 8859-1 encoded.
+     *
+     * @param filter if true, then entries in {@code p} with a key or value that is not a
+     *               String are filtered out. Otherwise, the caller guarantees {@code p}
+     *               only contains String keys and values.
      */
-    private static byte[] serializePropertiesToByteArray(Properties p) throws IOException {
+    private static byte[] serializePropertiesToByteArray(Properties p, boolean filter) throws IOException {
```

Comment:
I think we need more context as to why there are non-String system properties in use here.

Reply:
This parameter exists to allow a caller to pass in a `Properties` object that is guaranteed to have only String keys and so does not need the extra filtering done in this method. I'll refactor the code to make it clearer.

Reply:
Hopefully this makes it clearer: https://github.com/openjdk/jdk/pull/11513/commits/5c610798fe4eaed7efeb2eebcf1e5db47714caee

Reply:
Thanks. I initially assumed it was done to support non-String key/values but it's actually an optimization because the save properties are always Strings.